jgrapht-io/src/main/java/org/jgrapht/nio/dimacs/DIMACSImporter.java

```diff
@@ -159,19 +159,15 @@ public void importGraph(Graph<V, E> graph, Reader input)
     private class Consumers
     {
         private Graph<V, E> graph;
-        private Integer nodeCount;
-        private Map<Integer, V> map;
+        private final List<V> list = new ArrayList<>();
```

Comment:
Consider initializing this variable in the constructor.

Reply:
Alright.
Due to the design of the class (fields assigned with lambdas that refer to other fields) the field then cannot be final (like the graph field).

What do you think about, discarding the `Consumers` class and simplify creating the lambdas in two methods `createNodeCountConsumer(Graph<V,E>,List<V>)` and `createEdgeConsumer(Graph<V,E>,List<V>)` of the surrounding class?
This would be simpler than it is now. And because the parameter Graph and List are captured by the lambdas there is no need to model fields explicitly.

Reply:
These two consumers depend on the common vertex list. IIUC, while creating consumers in two separate methods, you'll still need to keep track of both graph and this common list in both lambdas. The advantage of the current code is that this dependency is very clear from the code, so I don't think it's worth it.

Reply:
Alright, if you prefer the current code, I'm fine with that and just leaf it as it is.